java.hints/src/org/netbeans/modules/java/hints/errors/VarCompDeclaration.java

```diff
@@ -0,0 +1,179 @@
+    }
+
+    @Override
+    public String getDisplayName() {
+        return NbBundle.getMessage(VarCompDeclaration.class, "FIX_VarCompDeclaration"); // NOI18N
+    }
+
+    public String getDescription() {
+        return NbBundle.getMessage(VarCompDeclaration.class, "FIX_VarCompDeclaration"); // NOI18N
+    }
+
+    @Override
+    public void cancel() {
+    }
+
+    private static final class FixImpl extends JavaFix {
+
+        CompilationInfo info;
+        TreePath path;
+
+        public FixImpl(CompilationInfo info, TreePath path) {
+            super(info, path);
+            this.info = info;
+            this.path = path;
+        }
+
+        @Override
+        protected String getText() {
+            return NbBundle.getMessage(VarCompDeclaration.class, "FIX_VarCompDeclaration"); // NOI18N
+        }
+
+        public String toDebugString() {
+            return NbBundle.getMessage(VarCompDeclaration.class, "FIX_VarCompDeclaration"); // NOI18N
+        }
+
+        @Override
+        protected void performRewrite(TransformationContext ctx) throws Exception {
+            TreePath statementPath = ctx.getPath();
+            Tree parent = statementPath.getParentPath().getLeaf();
+            List<? extends StatementTree> statements = null;
+            switch (parent.getKind()) {
+                case BLOCK:
+                    statements = ((BlockTree) parent).getStatements();
+                    break;
+                //Todo : Handle CASE scenario
+                /*case CASE:
+                    statements = ((CaseTree) parent).getStatements();
+                    break;*/
+                default:
+                    // Ignore other scenario
+                    break;
+            }
+            WorkingCopy wc = ctx.getWorkingCopy();
+            TreeMaker make = wc.getTreeMaker();
+            int pos = statements.indexOf(statementPath.getLeaf());
+            List<StatementTree> newStatements = new ArrayList<>();
+            if (pos > 0) {
+                newStatements.addAll(statements.subList(0, pos));
+            }
+
+            int current = 0;
+            for (current = pos; current < statements.size(); current++) {
+                StatementTree t = (StatementTree) statements.get(current);
+                if (t instanceof VariableTree) {
+                    VariableTree oldVariableTree = (VariableTree) t;
+                    VariableTree newVariableTree = make.Variable(
+                            oldVariableTree.getModifiers(),
+                            oldVariableTree.getName(),
+                            make.Type("var"), // NOI18N
+                            oldVariableTree.getInitializer()
+                    );
+                    newStatements.add(make.asReplacementOf(newVariableTree, oldVariableTree));
+                    
+                    // Check variable tree seperated with ","
+                    TokenSequence<JavaTokenId> tokenSequence = info.getTreeUtilities().tokensFor(t);
```

Comment:
@jlahoda shouldn't we have something in model (TreeUtilities ?), which could hide the tokensequence details and discover VariableTree groups ? Similar thing is done ad hoc in CasualDiff already, but may be needed for well-behaved lang model clients, too.

Reply:
Why not. I wouldn't focus too much on ability to reuse the implementation between TreeUtilities and CasualDiff (better if we can reuse, but CasualDiff is doing a lot of special stuff, so not sure if reuse is possible). It may be possible to detect field groups by looking at the modifiers of the variables (these may be shared among the VariableTrees, but I am not sure), or positions of the modifiers and/or types. But not sure offhand. Having an API method (with a test) would help to ensure that if changes are needed for the detection in the future, we can do them "easily".

Reply:
API changes added in TreeUtilities file (isTreeEndsWithComma)